pandas/core/series.py

```diff
@@ -3234,40 +3239,25 @@ def _try_cast(arr, take_fast_path):
     else:
         subarr = _try_cast(data, False)
 
-    # scalar like, GH
-    if getattr(subarr, 'ndim', 0) == 0:
-        if isinstance(data, list):  # pragma: no cover
-            subarr = np.array(data, dtype=object)
-        elif index is not None:
-            value = data
+    if subarr.ndim == 0 or is_scalar(data):
+        if index is None:
+            return subarr.item()
 
+        if subarr.ndim == 1:
+            # a scalar upcasted to 1-dimensional by maybe_cast_to_datetime()
```

Comment:
so would much rather fix this case *inside* maybe_cast_to_datetime

Reply:
My (limited) understanding is that it is impossible, basically because

```python
def broken_datetimeindex_roundtrip(dtix):
    scalar = dtix.values[0, ...]
    return pd.DatetimeIndex(np.array([scalar], dtype=scalar.dtype))
```
looses ``tz`` and ``freq``.
This in turn is probably not trivial at all to fix (or more honestly: I have no idea).

Reply:
> looses tz and freq.

Uhm... it probably makes perfect sense that it looses ``freq`` (but not ``tz``).